ios/brave-ios/Sources/BraveWallet/Crypto/Onboarding/RestoreWalletView.swift

```diff
@@ -271,7 +271,7 @@ struct RestoreWalletView: View {
     keyringStore.restoreWallet(
       words: recoveryWords,
       password: password,
-      isLegacyBraveWallet: recoveryWords.count == .legacyWalletRecoveryPhraseNumber
+      isLegacyEthSeedFormat: recoveryWords.count == .legacyWalletRecoveryPhraseNumber
```

Comment:
@StephenHeaps it seems like a bug to me:
For desktop `isLegacyEthSeedFormat` is set to true when there are 24 words and a checkbox is toggled, but for ios there is no checkbox
https://github.com/brave/brave-core/blob/master/components/brave_wallet_ui/page/screens/restore-wallet/restore-wallet.tsx#L206-L217

Reply:
@supermassive iOS has a button to toggle between 12 / 24 word fields showing:
https://github.com/brave/brave-core/blob/master/ios/brave-ios/Sources/BraveWallet/Crypto/Onboarding/RestoreWalletView.swift#L152-L174

![12-word](https://github.com/brave/brave-core/assets/5314553/0357c4ae-55db-4a53-a20b-2f943198ec71) | ![24-word](https://github.com/brave/brave-core/assets/5314553/b2704543-b775-44dc-8f36-6e68984ae47f)
--|--

Reply:
Desktop has another checkbox
So there are three outcomes:
12 words
24 words
24 words + legacy

![image](https://github.com/brave/brave-core/assets/168356/0942a382-3fa6-4ed2-93f2-a89f8ea63f84)

Reply:
It's actually a bigger problem as all mnemonics with 24 words are considered "legacy" now for ios. Which means restoring mnemonic on another wallet app will produce different set of ETH addresses

Reply:
@nuo-xu can you take a look at this tomorrow? Looks like toggle was [dropped when v2 UI was integrated](https://github.com/brave/brave-ios/pull/7746), previous would [show toggle when 24-words was selected](https://github.com/brave/brave-ios/blob/e72f2d7bf5f08bead1795e3e2f03016c52a42943/Sources/BraveWallet/Crypto/Onboarding/RestoreWalletView.swift#L105). `isBraveLegacyWallet` state value there, but no toggle for it and v2 UI is instead [using word count](https://github.com/brave/brave-core/blob/9f204d486f58a098f9687464150e24df0bb2160e/ios/brave-ios/Sources/BraveWallet/Crypto/Onboarding/RestoreWalletView.swift#L33-L35).

Reply:
[issue ](https://github.com/brave/brave-browser/issues/38123) created. will fix asap.